IfcElectricFlowTreatmentDevice Rail Import BIMQ 2021-07-21T06:51:56.6

jmirtsch opened the pull request:
Rail properties for IfcElectricFlowTreatmentDevice

jmirtsch committed:
IfcElectricFlowTreatmentDevice Rail Import BIMQ 2021-07-21T06:51:56.6885238Z

jmirtsch committed:
Explicit IfcLabel PrimaryDataType

jmirtsch commented on IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SencondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SencondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SecondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```

jmirtsch commented on IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/Documentation.md:
```diff
@@ -0,0 +1 @@
+The nominal frequency at scondary side
```
```suggestion
The nominal frequency at secondary side
```

jmirtsch commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml:
```diff
@@ -0,0 +1,12 @@
+		<DocProperty xsi:nil="true" href="NominalPower_2pu16hiH1CqQv04BNgxAph" />
+		<DocProperty xsi:nil="true" href="RatedVoltage_0KkDpffYz8L86gzG31TCSi" />
+		<DocProperty xsi:nil="true" href="PrimaryFrequency_3bDvHjEfEAwV9_ElCeTdI" />
+		<DocProperty xsi:nil="true" href="SencondaryFrequency_272XQ9Dt16FOR1miSL0g" />
```
```suggestion
		<DocProperty xsi:nil="true" href="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" />
```

jmirtsch commented on IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Type_1CfKTChb3WvEVYC2jfIpk" Name="Type" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```
```suggestion
<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FiltersType_1CfKTChb3WvEVYC2jfIpk" Name="FiltersType" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```

jmirtsch commented on IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_ElectricFlowTreatmentDeviceTypeElectronicFilter/DocPropertySet.xml:
```diff
@@ -0,0 +1,12 @@
+		<DocProperty xsi:nil="true" href="RatedVoltage_0KkDpffYz8L86gzG31TCSi" />
+		<DocProperty xsi:nil="true" href="PrimaryFrequency_3bDvHjEfEAwV9_ElCeTdI" />
+		<DocProperty xsi:nil="true" href="SencondaryFrequency_272XQ9Dt16FOR1miSL0g" />
+		<DocProperty xsi:nil="true" href="Type_1CfKTChb3WvEVYC2jfIpk" />
```
```suggestion
		<DocProperty xsi:nil="true" href="FiltersType_1CfKTChb3WvEVYC2jfIpk" />
```

fhulin committed:
Apply suggestions from code review

Co-authored-by: Jon Mirtschin <jonm@geometrygym.com>

SergejMuhic commented on IFC4x3/Properties/n/NominalCurrent_1Sia_y4DvB595bUBLM1Dsb/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalCurrent_1Sia_y4DvB595bUBLM1Dsb" Name="NominalCurrent" UniqueId="5cb24fbc-10de-4b14-9165-78b55604dda5" PrimaryDataType="IfcElectricCurrentMeasure" />
```
Reuse

fhulin replied:
Done. Commit: cc2999beef494e17b3c8aa19f47045f89c8c1bb8

fhulin replied:
Erased in commit : 7215357c22b15b8f871b30c0abe5495916bdbcf9

SergejMuhic commented on IFC4x3/Properties/n/NominalPower_2pu16hiH1CqQv04BNgxAph/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NominalPower_2pu16hiH1CqQv04BNgxAph" Name="NominalPower" UniqueId="b3e011ab-b110-4cd1-ae40-10b5eaecaceb" PrimaryDataType="IfcPowerMeasure" />
```
Reuse

fhulin replied:
There is no existing generic NominalPower property. So keep the one we add with a generic definition from IEC.

fhulin replied:
> * constant documentation
> 
>     * pednum documentation

Documentation is done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

fhulin replied:
Already using the same NominalPower that have been merged

SergejMuhic commented on IFC4x3/Properties/p/PrimaryFrequency_3bDvH$jEfEAwV9_ElCeTdI/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PrimaryFrequency_3bDvHjEfEAwV9_ElCeTdI" Name="PrimaryFrequency" UniqueId="e537947f-b4ea-4e2b-a7c9-f8ebcca1d9d2" PrimaryDataType="IfcFrequencyMeasure" />
```
Reuse

fhulin replied:
Done : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

SergejMuhic commented on IFC4x3/Properties/r/RatedVoltage_0KkDpffYz8L86gzG31TCSi/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RatedVoltage_0KkDpffYz8L86gzG31TCSi" Name="RatedVoltage" UniqueId="14b8dce9-a62f-4854-81aa-f500c174c72c" PrimaryDataType="IfcElectricVoltageMeasure" />
```
Reuse

fhulin replied:
Done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

SergejMuhic commented on IFC4x3/Properties/s/SencondaryFrequency_272XQ9Dt16FOR1$miSL0g$/DocProperty.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SecondaryFrequency_272XQ9Dt16FOR1miSL0g" Name="SecondaryFrequency" UniqueId="870a1689-3770-463d-86c1-ff0b1c540abf" PrimaryDataType="IfcFrequencyMeasure" />
```
Reuse

fhulin replied:
Done. Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

SergejMuhic commented on IFC4x3/Properties/t/Type_1CfK$TChb3WvEVYC2jfIpk/DocProperty.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FiltersType_1CfKTChb3WvEVYC2jfIpk" Name="FiltersType" UniqueId="4ca54fdd-32b9-4383-939f-88c0ada52cee" PropertyType="p_enumeratedvalue" PrimaryDataType="IfcLabel">
```
Are there many filters? I would prefer `FilterType` even if there are many.

fhulin replied:
Agree! Commit : cc2999beef494e17b3c8aa19f47045f89c8c1bb8

SergejMuhic reviewed (changes requested):
- constant documentation
- pednum documentation

fhulin committed:
From Comments

fhulin committed:
Erasing Nominal Current

fhulin commented:
@SergejMuhic All issues are now solved.

AlexBrad1eyCT committed:
Merge branch 'develop' into RailProperties_IfcElectricFlowTreatmentDevice

AlexBrad1eyCT committed:
Delete DocProperty.xml

AlexBrad1eyCT committed:
Delete Documentation.md

AlexBrad1eyCT committed:
Name change for FilterType

Made the name more specific because it is connected to a particular filter enumeration for electronic filters, leaving in its current state would hinder efforts to catagorise other objects that are known as "Filters"

AlexBrad1eyCT reviewed (approved):
LFTM!

AlexBrad1eyCT committed:
Merge branch 'develop' into RailProperties_IfcElectricFlowTreatmentDevice
